docs/proposals/envvar-configmap.md

```diff
@@ -0,0 +1,182 @@
+# ConfigMaps as environment variables
+
+## Goal
+
+Populating environment variables of a container from an entire ConfigMap.
+
+## Design Points
+
+A container can specify a set of existing ConfigMaps to populate environment variables.
+
+There needs to be an easy way to isolate the variables introduced by a given
+ConfigMap. The contents of a ConfigMap may not be known in advance and it may
+be generated by someone or something else. Services may provide binding
+information via a ConfigMap. If you have a common service with multiple
+instances like a Message Queue or Database, there needs to be a way to
+uniquely identify and prevent collision when combining these ConfigMaps.
+
+## Proposed Design
+
+Containers can specify a set of sources that are consumed as environment
+variables. One such source is a ConfigMap.
+Each key defined in the ConfigMap's `Data` object must be a "C" identifier. If
+an invalid key is present, the container will fail to start.
+
+Environment variables are currently defined by services or the `Env` object in
+a container, where the `Env` takes precedence.  The introduction of ConfigMaps
+adds a third possibility. To prevent any change in behavior, the `Env` object
+will still override any environment variable introduced by a ConfigMap.
+A ConfigMap is allowed to override variables defined by services.  Variable
+references defined by an `EnvVar` struct can be resolved by values defined in
+other `EnvVar`s or those introduced by ConfigMaps and services.
+
+To prevent collisions amongst multiple ConfigMaps, each defined ConfigMap can
+have an optional associated prefix that is prepended to each key in the
+ConfigMap.  Prefixes must be a "C" identifier.
+
+### Kubectl updates
+
+The describe command will display the configmap name that have been defined as
```

Comment:
name does not uniquely identify which configmap was used.  You would need resource version and UID to confirm a match between the pod and a configmap object.